validator/client/validator_metrics.go

```diff
@@ -51,72 +51,64 @@ func (v *validator) LogValidatorGainsAndLosses(ctx context.Context, slot uint64)
 		return err
 	}
 
-	missingValidators := make(map[[48]byte]bool)
-	for _, val := range resp.MissingValidators {
-		missingValidators[bytesutil.ToBytes48(val)] = true
+	if v.emitAccountMetrics {
+		for _, missingPubKey := range resp.MissingValidators {
+			fmtKey := fmt.Sprintf("%#x", missingPubKey[:])
+			validatorBalancesGaugeVec.WithLabelValues(fmtKey).Set(0)
+		}
 	}
 
 	included := 0
 	votedSource := 0
 	votedTarget := 0
 	votedHead := 0
-
-	reported := 0
-	for _, pkey := range pubKeys {
-		pubKey := fmt.Sprintf("%#x", pkey[:8])
-		log := log.WithField("pubKey", pubKey)
-		fmtKey := fmt.Sprintf("%#x", pkey[:])
-		if missingValidators[bytesutil.ToBytes48(pkey)] {
-			if v.emitAccountMetrics {
-				validatorBalancesGaugeVec.WithLabelValues(fmtKey).Set(0)
-			}
-			continue
-		}
+	prevEpoch := (slot / params.BeaconConfig().SlotsPerEpoch) - 1
```

Comment:
How about slot 0? I believe we have a helper func called helpers.PrevEpoch that checks for underflow

Reply:
This code doesn't run in epoch 0, but I'll use the helper func.

Reply:
lets make a small helper then in this file that checks for underflow

Reply:
Added a comment to detail